code/system/application/perception/src/detectlane/detectlane.cpp

```diff
@@ -379,99 +439,183 @@ void DetectLane::setUp()
   m_K = Eigen::MatrixXd(m_nRegions,1);
   m_M = Eigen::MatrixXd(m_nRegions,1);
 
-  m_leftTransformationMatrix = readreadThreeByThreeMatrix(
-          "/home/batko/Desktop/leftTransformationMatrix.txt");
-  m_rightTransformationMatrix = readreadThreeByThreeMatrix(
-          "/home/batko/Desktop/rightTransformationMatrix.txt");
 
+  m_leftTransformationMatrix = ReadMatrix(
+          "/opt/opendlv/share/opendlv/tools/vision/projection/leftCameraTransformationMatrixWarped.csv",3,3);
+  std::cout<<m_leftTransformationMatrix;
+  std::cout<<"\n------\n----\n---\n";
+  m_rightTransformationMatrix = ReadMatrix(
+          "/opt/opendlv/share/opendlv/tools/vision/projection/rightCameraTransformationMatrixWarped.csv",3,3);
+
+  m_scale << m_width / (double)m_outputWidth, m_height / (double)m_outputHeight, 1;
+ 
+  if(false)
+  {
+    std::cout<<"Input size\n";
+    std::cout<<m_width<< " " << m_height << std::endl;
+    std::cout<<"Output size\n",
+    std::cout << m_outputWidth << " " << m_outputHeight<<std::endl;
+    std::cout<<"Left camera regions:\n";
+    std::cout<<m_leftCameraRegions;
+    std::cout<<std::endl;
+    std::cout<<"Right camear regions:\n";
+    std::cout<<m_rightCameraRegions<<std::endl;
+    std::cout<<std::endl;
+    std::cout<<"Left camera transformation matrix\n";
+    std::cout<<m_leftTransformationMatrix<<std::endl;
+    std::cout<<std::endl;
+    std::cout<<"Right camera transformation matrix\n";
+    std::cout<<m_rightTransformationMatrix<<std::endl;
+    std::cout<<std::endl;
+    std::cout<<"Left camera warp points :\n";
+    for(int i = 0; i < 4; i++)
```

Comment:
There should always be brackets on statements, since this is a common cause of bugs.